src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Installation/Checksum/SHA256DirectoryChecksumCalculator.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.Performance.Toolkit.Plugins.Core;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Installation
+{
+    /// <summary>
+    ///     Represents a checksum calculator that uses SHA256.
+    /// </summary>
+    public sealed class SHA256DirectoryChecksumCalculator
+        : IDirectoryChecksumCalculator
+    {
+        /// <inheritdoc />
+        public async Task<string> GetDirectoryChecksumAsync(string directory)
+        {
+            return await HashUtils.GetDirectoryHashAsync(directory);
```

Comment:
You might want to move the impl here so it's obvious that it MUST be a SHA256 (given the class name)

Reply:
Updated